### PR TITLE
Avoid Complex objects on Span tags

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR is for avoiding Complex objects on Span tags and enables a new api for that `UnsafeSetTag`.

The complex objects are transformed by the new `GetValidValue` func.

### CODE REVIEW

File: instrumentation/testing/testing.go [hunk 1]
```diff
@@ -214,7 +214,11 @@ func (test *Test) end() {
 			instrumentation.Logger().Printf("CodePath in parallel test is not supported: %v\n", test.t.Name())
 			coverage.RestoreCoverageCounters()
 		} else if cov := coverage.EndCoverage(); cov != nil {
-			test.span.SetTag(tags.Coverage, *cov)
+			if sp, ok := test.span.(tracer.Span); ok {
```
Comment (drodriguezhdez) [inline instrumentation/testing/testing.go:217]: Here it is checked if the span is our span, tag is gonna be set using `unsafe`, which means tag is not gonna be escaped?

Comment (tonyredondo) [reply instrumentation/testing/testing.go:217]: That's right the idea is that we know the coverage object is "safe" so if we detect our span, we ensure that we write the object without any modification. But if we are using other tracer we call the normal opentracing Api, that should be followed by the tracer.

Comment (drodriguezhdez) [reply instrumentation/testing/testing.go:217]: Is there any case where we should use the "escaped" mode in our spans? Or only it is needed when the user is adding custom spans?

Comment (tonyredondo) [reply instrumentation/testing/testing.go:217]: When the user is adding custom spans, but we fallback to the Unescape mode if we don't detect a complex object, so its safe call the "escaped" version. Also there are some "helper" methods inside the api, that call directly to the api version of the SetTag (in our case the "escaped" version) so we need to play nice with the api.

**Pull request merged**